### ISSUE TITLE
Provide a consistent interface for marking up Will messages (markdown)

### ISSUE THREAD

**Issue by skoczen** [skoczen/will#310]
This comes off discussion in #308, around the following idea: we should be able to make rich, readable Will messages with basic formatting, html links, etc without having to use platform dependent syntax.

The first thought on implementation would be to use Github-flavored Markdown, as it's widely in use, consistent, and full-featured.

One possible implementation is mistune:

@netjunki: 

>   I think Markdown is a good common markup for this use case. Not sure about the best way to go about implementing the chat system specific rendering. Though [mistune](https://github.com/lepture/mistune) looks like it provides a [promising feature set in this area](https://github.com/lepture/mistune#renderer).

Discussion and issue tracking here - let's get a full set of cases where folks would like to add formatting, and make sure we can cover as many as possible!

**Comment by netjunki** [comment]
The most interesting challenge here to me is how would you deal with the following markdown in different platforms:

    | Command | Description |
    | --- | --- |
    | git status | List all new or modified files |
    | git diff | Show file differences that haven't been staged |

In Hipchat we'd want that to become an html table.
In FlowDock while it supports markdown doesn't support tables so we'd want to render this out somehow as a text table and then drop it in
In Slack we'd have to render it out as text the same way and drop it in

**Comment by skoczen** [comment]
Yeah, or I'd say there's probably a middle ground - just support basic markdown, and then provide a raw HTML option for platforms that support it.
